src/HttpCache/EventListener/AddTagsListener.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\HttpCache\EventListener;
+
+use ApiPlatform\Core\Api\IriConverterInterface;
+use ApiPlatform\Core\HttpCache\ResourceManagerInterface;
+use ApiPlatform\Core\Util\RequestAttributesExtractor;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+
+/**
+ * Sets the list of resources' IRIs included in this response in the "Cache-Tags" HTTP header.
+ *
+ * The "Cache-Tags" is used because it is supported by CloudFlare.
+ *
+ * @see https://support.cloudflare.com/hc/en-us/articles/206596608-How-to-Purge-Cache-Using-Cache-Tags-Enterprise-only-
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+final class AddTagsListener
+{
+    private $resourceManager;
+    private $iriConverter;
+
+    public function __construct(ResourceManagerInterface $resourceManager, IriConverterInterface $iriConverter)
+    {
+        $this->resourceManager = $resourceManager;
+        $this->iriConverter = $iriConverter;
+    }
+
+    /**
+     * Adds the "Cache-Tags" and clears the manager.
+     *
+     * @param FilterResponseEvent $event
+     */
+    public function onKernelResponse(FilterResponseEvent $event)
+    {
+        $request = $event->getRequest();
+        if (
+            !$request->isMethodCacheable()
+            || !$attributes = RequestAttributesExtractor::extractAttributes($request)
+        ) {
```

Comment:
To address @jderusse's [comment](https://github.com/api-platform/core/pull/952#issuecomment-288318323), we'd add another condition here:

```php
!in_array($response->getStatusCode(), [
    Response::HTTP_OK,
    Response::HTTP_NO_CONTENT,
])
```